### ISSUE TITLE
Tracking issue: If/else (ternary operator)

### ISSUE THREAD

**Issue by casey** [casey/just#696]
I propose adding an `if` construct, restricted by the parser to equality tests:

```
foo := if A == B { THEN } else { ELSE }
```

`A`, `B`, `THEN`, and `ELSE` are all arbitrary expressions. If `A` equal to `B` then it will evaluate as `THEN`, otherwise it will evaluate as `ELSE`.

The reason I propose adding `if`, restricted to equality testing, instead of a more general `if`, is because if, in some future in which Just get's a type system, I would like the condition, `A == B` above, of `if` to be a bool. By restricting `if` to equality testing, a future enhancement which adds boolean and relaxes `if's condition to be any expression would not be a breaking change.

I added the `good first issue` tag to this. It's tough, but I think doable for someone that's interested in learning how the lexer and parser work. The only complication I see is making sure that `if` isn't always parsed as a keyword, so that having a recipe named `if` doesn't become an error.

(See prior discussion in #262 and #674.)

**Comment by mortoray** [comment]
Perhaps an `if-equal` would also be useful and also be forwards compatible. Making it string based would also include the if-empty logic in it's coverage.

foo := if-equal {{bar}} "hi" { first } else { other }
nothing := if-equal {{value}} "" {first} else { other }

**Comment by kenden** [comment]
@mortoray  It would be nice to have a `if-equal`, but that syntax it not so nice... It makes for some hard to read one-liners.

**Comment by casey** [comment]
I think if `if-equal` were introduced, it might be good to give it the fixed syntax `if A == B { THAN } else { ELSE }`. Since I can't imagine any syntax other than `==` for equality, this would also be forward compatible.

**Comment by mortoray** [comment]
I agree. It would also satisfy the if-empty need, `if A == "" { THEN } else { ELSE }`

**Comment by casey** [comment]
I think adding `if A == B ...` is definitely the right way to go. I rewrite the original post and removed mention of `if-empty`.

**Comment by casey** [comment]
It is done! (#714) I added both equality and inequality conditionals:

```make
foo := if "a" == "a" { "xyz" } else { "abc" }

bar := if "a" != "b" { "xyz" } else { "abc" }
```

They can be used in any context where an expression was valid, so both at the top level and inside recipe interpolations.

The can only be used in expression contexts, so they can't be used to conditionally enable different parts of a justfile, like an `#ifdef`

I just released [v0.8.2](https://github.com/casey/just/releases/tag/v0.8.2) that includes the new feature.

Thanks everyone for requesting this feature and discussing how it should work!
